plugins/performance-lab/includes/admin/plugins.php

```diff
@@ -54,61 +54,94 @@ function perflab_query_plugin_info( string $plugin_slug ) {
 		)
 	);
 
+	$has_errors = false;
+	$plugins    = array();
+
 	if ( is_wp_error( $response ) ) {
-		return new WP_Error(
-			'api_error',
-			sprintf(
-				/* translators: %s: API error message */
-				__( 'Failed to retrieve plugins data from WordPress.org API: %s', 'performance-lab' ),
-				$response->get_error_message()
-			)
+		$plugins[ $plugin_slug ] = array(
+			'error' => array(
+				'code'    => 'api_error',
+				'message' => sprintf(
+					/* translators: %s: API error message */
+					__( 'Failed to retrieve plugins data from WordPress.org API: %s', 'performance-lab' ),
+					$response->get_error_message()
+				),
+			),
 		);
+
+		$has_errors = true;
 	}
 
 	// Check if the response contains plugins.
-	if ( ! ( is_object( $response ) && property_exists( $response, 'plugins' ) ) ) {
-		return new WP_Error( 'no_plugins', __( 'No plugins found in the API response.', 'performance-lab' ) );
+	if ( ! $has_errors && ! ( is_object( $response ) && property_exists( $response, 'plugins' ) ) ) {
+		$plugins[ $plugin_slug ] = array(
+			'error' => array(
+				'code'    => 'no_plugins',
+				'message' => __( 'No plugins found in the API response.', 'performance-lab' ),
+			),
+		);
+
```

Comment:
I think for both of these scenarios we need to set the error for every single plugin of ours. In other words, iterate through the list from `perflab_get_standalone_plugins()` and set the same error for every plugin's slug.

Otherwise, this can cause a problem where the API failing would cause the request to be issued for every single plugin, which would defeat the purpose of the transient cache and the combination of those plugin requests into a single API request (see #1542). This would effectively fix #1691, and since this PR is touching the exact same logic, I wonder whether we should go straight to a solution that addresses both issues.

Reply:
To make things clear as its getting confusing please answer these below question or my understanding of the solution needed.

- If error is encounterd at line `57` and `76` we should add the error to the all `perflab_get_standalone_plugins` plugins so that if any one of the plugin is requested it should return error?

Reply:
Correct. That's because a failing API response is by definition an error that affects all plugins.